tox.ini

```diff
@@ -245,7 +245,10 @@ commands =
     {[base]pip_install} acme certbot certbot-dns-rfc2136 certbot-ci
     pytest certbot-ci/certbot_integration_tests/rfc2136_tests \
         --acme-server=pebble --dns-server=bind \
-        --numprocesses=1
+        --numprocesses=1 \
+        --cov=acme --cov=certbot --cov=certbot_dns_rfc2136 --cov-report= \
+        --cov-config=certbot-ci/certbot_integration_tests/.coveragerc
+    coverage report --include 'certbot-dns-rfc2136/*' --show-missing --fail-under=87
```

Comment:
Since the DNS plugin relies heavily on certbot core functionalities + a dedicated API designed in certbot itself, maybe it makes sense to also add a coverage rule to the certbot project here.

What do you think of this?

Reply:
Do you mean the base `DNSAuthenticator` class?

The coverage of `certbot/*` overall is quite low in this certbot-dns-rfc2136 integration test: 46%. Does tracking that achieve a useful result? Would a more useful result be to track coverage in specific modules, like `dns_common`?

Reply:
Sorry I missed that.

Between covering specifically `dns_common` or the entire `certbot` module, I do not know what is better.

I would say that any coverage value is relevant, what matters it the dynamic of it. What would make even more sense would be to refresh regularly the threshold to ensure that dynamic is to progressively increase the coverage. We can expect that dns plugins are triggering a certbot logic which is slightly different that the major case (HTTP-01 challenges), so I would say for simplicity, that we could stay on an coverage assertion over the entire `certbot/*`, that is less likely to lose its relevance silently if we heavily refactor how DNS plugins are implemented in Certbot.

Reply:
>What would make even more sense would be to refresh regularly the threshold to ensure that dynamic is to progressively increase the coverage

Agreed. Maybe it would be nice if eventually there was something to check the CI output for large disparities between actual and `--fail-under`.